src/java/neroxis/map/SCMap.java

```diff
@@ -376,6 +382,257 @@ public void addCubeMap(CubeMap cubeMap) {
+
+        FloatMask[] texturesMasks = getTextureMasksScaled(symmetrySettings);
+        FloatMask oldLayer1 = texturesMasks[0];
+        FloatMask oldLayer2 = texturesMasks[1];
+        FloatMask oldLayer3 = texturesMasks[2];
+        FloatMask oldLayer4 = texturesMasks[3];
+        FloatMask oldLayer5 = texturesMasks[4];
+        FloatMask oldLayer6 = texturesMasks[5];
+        FloatMask oldLayer7 = texturesMasks[6];
+        FloatMask oldLayer8 = texturesMasks[7];
+
+        setTextureMasksLow(new BufferedImage(newMapBoundsSize, newMapBoundsSize, BufferedImage.TYPE_INT_ARGB));
+        setTextureMasksHigh(new BufferedImage(newMapBoundsSize, newMapBoundsSize, BufferedImage.TYPE_INT_ARGB));
+
+        oldLayer1.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer2.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer3.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer4.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer5.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer6.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer7.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+        oldLayer8.min(0f).max(1f).setSize(resizeCurrentMapContentTo);
+
+        FloatMask layer1 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer1.addWithOffset(oldLayer1, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer2 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer2.addWithOffset(oldLayer2, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer3 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer3.addWithOffset(oldLayer3, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer4 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer4.addWithOffset(oldLayer4, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer5 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer5.addWithOffset(oldLayer5, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer6 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer6.addWithOffset(oldLayer6, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer7 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer7.addWithOffset(oldLayer7, locToPutCenterOfCurrentMapContent, true);
+        FloatMask layer8 =  new FloatMask(newMapBoundsSize, new Random().nextLong(), symmetrySettings);
+        layer8.addWithOffset(oldLayer8, locToPutCenterOfCurrentMapContent, true);
+
+        setTextureMasksLowScaled(layer1, layer2, layer3, layer4);
+        setTextureMasksHighScaled(layer5, layer6, layer7, layer8);
```

Comment:
Why are the textureMasks handled any differently they are just bufferedImages as well

Reply:
I tried handling them like the others first, but the result of that was that when the texture masks were used to be only part of the new image, the new texture masks were in the bottom right corner of the image rather than in the desired location.

Reply:
i suspect then that is because the texture mask size was different from the map size. As it is done currently the size could still be off as you are assuming the size of the texture masks was the same size as the map which is not always the case. Also I don;t think the scaled float masks should be used as they can cause changes in the textures due to scaling issues. Ultimately better to figure out the sizing issues with the imageutils

Reply:
I had spent a while trying to figure this issue out better for the populator, and this is the best system for this that we implemented.  It does work consistently, although some gpg textures can be a bit rough around the edges... if you know how to do a better solution for this, feel free to implement it.